### DISCUSSION TITLE
HTTP version

### DISCUSSION THREAD

**Discussion by robin-thoene** [open-telemetry/opentelemetry-rust#1627] [Q&A] [answered]
Hey all,

I am currently experimenting with OTEL using `axum` as API framework. The current version of axum uses the 1.x.x version of the `http` crate, but here the version is set to 0.2 in the workspace.

This causes some errors when I try to extract the parent span using an incoming request.

I was wondering if there is a reason for the version 0.2 being used and/or if it is safe to update the version to 1.x.x here as well (if so I would love to create and issue and implement it).

**Comment by TommyCpp** [comment, marked as answer]
We have an open issue regarding upgrade to HTTP 1.0 https://github.com/open-telemetry/opentelemetry-rust/issues/1427

The current blocker is tonic is still depending on HTTP 0.2 and that forced us to hold the upgrade for now

**Reply by robin-thoene** [reply]
Ah I see thank you, maybe I can help tonic with it 😊

**Reply by xmakro** [reply]
tonic was upgraded to 0.12
